testing/test_warnings.py

```diff
@@ -516,6 +516,24 @@ def test_hidden_by_system(self, pytester: Pytester, monkeypatch) -> None:
         result = pytester.runpytest_subprocess()
         assert WARNINGS_SUMMARY_HEADER not in result.stdout.str()
 
+    def test_recwarn(self, pytester: Pytester, monkeypatch) -> None:
+        """
+        Check that recwarn can capture DeprecationWarning by default
+        without custom filterwarnings (see #8666).
+        """
+        pytester.makepyfile(
```

Comment:
I think this test should be inside test_recwarn.py?

also I don't think `pytester` is needed here -- this can be tested directly (?)

the testsuite is failing because this test function name matches `test_recwarn.py`

Reply:
Good points, all done.

I initially used pytester to ensure our own warning filters wouldn't interfere with the test, but in the end I'm not sure it is needed after all.